packages/driver/cypress/integration/e2e/origin/commands/waiting.spec.ts

```diff
@@ -1,33 +1,341 @@
+        })
+      }).as('foo')
+
+      cy.once('command:retry', () => xhrGet('/foo'))
+
+      cy.origin('http://www.foobar.com:3500', () => {
+        cy.wait('@foo.request').then((xhr) => {
+          expect(xhr.request.url).to.include('/foo')
+          expect(xhr.response).to.be.undefined
+        })
+      })
+    })
+
+    it('waits for the route alias to have a response', () => {
+      const response = { foo: 'foo' }
+
+      cy.intercept('/foo', (req) => {
+        // delay the response to ensure the wait will wait for response
+        req.reply({
+          delay: 100,
+          body: response,
+        })
+      }).as('foo')
+
+      cy.once('command:retry', () => xhrGet('/foo'))
+
+      cy.origin('http://www.foobar.com:3500', { args: { response } }, ({ response }) => {
+        cy.wait('@foo.response').then((xhr) => {
+          expect(xhr.request.url).to.include('/foo')
+          expect(xhr.response?.body).to.deep.equal(response)
+        })
+      })
+    })
+
+    it('waits for the route alias (aliased through \'as\')', () => {
+      const response = { foo: 'foo' }
+
+      cy.intercept('/foo', response).as('foo')
+
+      cy.origin('http://www.foobar.com:3500', { args: { response } }, ({ response }) => {
+        cy.then(() => window.xhrGet('/foo'))
+        cy.wait('@foo').its('response.body').should('deep.equal', response)
+      })
+    })
+
+    it('waits for the route alias (aliased through req object)', () => {
+      const response = { foo: 'foo' }
+
+      cy.intercept('/foo', (req) => {
+        req.reply(response)
+        req.alias = 'foo'
+      })
+
+      cy.origin('http://www.foobar.com:3500', { args: { response } }, ({ response }) => {
+        cy.then(() => window.xhrGet('/foo'))
+
+        cy.wait('@foo').its('response.body').should('deep.equal', response)
+      })
+    })
+
+    it('has the correct log properties', () => {
+      const response = { foo: 'foo' }
+
+      cy.intercept('/foo', response).as('foo')
+
+      cy.origin('http://www.foobar.com:3500', { args: { response } }, ({ response }) => {
+        cy.then(() => window.xhrGet('/foo'))
+        cy.wait('@foo').its('response.body').should('deep.equal', response)
+      })
+
+      cy.shouldWithTimeout(() => {
+        const actualLog = Cypress._.pick(findCrossOriginLogs('wait', logs, 'localhost'),
+          ['name', 'referencesAlias', 'aliasType', 'type', 'instrument', 'message'])
+
+        const expectedLog = {
+          name: 'wait',
+          referencesAlias: [{ name: 'foo', cardinal: 1, ordinal: '1st' }],
+          aliasType: 'route',
+          type: 'parent',
+          instrument: 'command',
+          message: '',
+        }
+
+        expect(actualLog).to.deep.equal(expectedLog)
+      })
+    })
+
+    it('doesn\'t log when log: false', () => {
+      const response = { foo: 'foo' }
+
+      cy.intercept('/foo', response).as('foo')
+
+      cy.origin('http://www.foobar.com:3500', { args: { response } }, ({ response }) => {
```

Comment:
does the response need to be passed in to `cy.origin` here?

Reply:
No, it doesn't, updated [957b80e](https://github.com/cypress-io/cypress/pull/21579/commits/957b80e549ec301eb382a3cef29a232288013c58)